src/test/java/by/exadel/internship/formTest/FormTest.java

```diff
@@ -2,75 +2,151 @@
+        URI uri = UriComponentsBuilder.fromPath("/forms")
+                .queryParam("internshipId", "68a051d7-6d82-4879-b0a3-1340e14db54d")
+                .build().toUri();
+        MvcResult result = getResult(HttpMethod.GET, uri, status().isOk());
+        String content = result.getResponse().getContentAsString();
+        List<FormFullDTO> fos = objectMapper.readValue(content, new TypeReference<>() {
+        });
+        assertEquals(fos.size(), 1);
+    }
+
+    @Test
+    public void checkWhetherStatusIsUpdated() throws Exception {
+        URI uri = UriComponentsBuilder.fromPath("/forms/updateStatus")
+                .queryParam("formId", "7c0811d5-354b-4ebb-a65c-0b54efc53a80")
+                .queryParam("status", FormStatus.NOT_MATCHED)
+                .build().toUri();
+        getResult(HttpMethod.PUT, uri, status().isOk());
+        UUID uuid = UUID.fromString("7c0811d5-354b-4ebb-a65c-0b54efc53a80");
+        Form form = formRepository.findById(uuid).orElseThrow(() -> new NotFoundException("Form with uuid = " + uuid +
+                " Not Found in DB", "form.uuid.invalid"));
+        assertEquals(form.getFormStatus(), FormStatus.NOT_MATCHED);
+    }
+
+
+    @Test
+    public void givenFormList_WhenDeleteForm_ThenCheckListSizeWithFlagTrue() throws Exception {
+
+        UUID id = uuidOfCreatedAndPostTestDataForm();
+        getResult(HttpMethod.DELETE, URI.create("/forms/" + id), status().isOk());
+        List<Form> forms = formRepository.findAllByDeletedTrue();
+        assertEquals(forms.size(), 1);
+    }
+
+    @Test
+    public void givenForm_WhenDelete_ThenCheck_WhetherFormIsDeleted() throws Exception {
+
+        UUID id = uuidOfCreatedAndPostTestDataForm();
+        getResult(HttpMethod.DELETE, URI.create("/forms/" + id), status().isOk());
+        Form form = formRepository.findByIdAndDeletedTrue(id)
+                .orElseThrow(() -> new NotFoundException("Form with uuid = " + id +
+                        " Not Found in DB", "form.uuid.invalid"));
+        assertTrue(form.isDeleted());
+    }
+
+
+    @Test
+    public void when_RestoreForm_Expect_FlagIsFalse() throws Exception {
+        UUID id = uuidOfCreatedAndPostTestDataForm();
+        getResult(HttpMethod.DELETE, URI.create("/forms/" + id), status().isOk());
+        getResult(HttpMethod.PUT, URI.create("/forms/" + id + "/restore"), status().isOk());
+        Form form = formRepository.findByIdAndDeletedFalse(id)
+                .orElseThrow(() -> new NotFoundException("Form with uuid = " + id +
+                        " Not Found in DB", "form.uuid.invalid"));
+        assertFalse(form.isDeleted());
+    }
+
+    private UUID uuidOfCreatedAndPostTestDataForm() throws Exception {
+        MockMultipartFile testForm = new MockMultipartFile("form", "form",
+                MediaType.APPLICATION_JSON_VALUE,
+                ("{\"firstName\": \"testName\",\"city\": {\"id\":\"69e6b47a-4c3d-4207-ac2d-801d9eda7ff1\",\"name\":\"Lviv\"}," +
```

Comment:
Is this json and json from **givenFormWithFile_checkTestData** the same?  If yes, move it to one variable/method

Reply:
Done